lib/pure/algorithm.nim

```diff
@@ -369,3 +369,113 @@ when isMainModule:
   for i in 0 .. high(arr1):
     assert arr1.reversed(0, i) == arr1.reversed()[high(arr1) - i .. high(arr1)]
     assert arr1.reversed(i, high(arr1)) == arr1.reversed()[0 .. high(arr1) - i]
+
+
+proc rotate*[T](arg: var openarray[T]; first, middle, last: int): int {.discardable.} =
```

Comment:
Firstly, I avoid .discardable return types, they mess with type inference.
Secondly, it's not even documented what it returns.